### DISCUSSION TITLE
Storybook Template

### DISCUSSION THREAD

**Discussion by jigz** [vercel/vercel#4138] [General] [answered]
Having trouble deploying storybook, would be nice if there was a framework preset/template for storybook.. Am I missing something? 
```
// package.json
{
  "name": "components",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start",
    "storybook": "start-storybook -p 6006",
    "build-storybook": "build-storybook"
  },
  "dependencies": {
    "next": "9.3.5",
    "react": "16.13.1",
    "react-dom": "16.13.1"
  },
  "devDependencies": {
    "@babel/core": "^7.9.0",
    "@storybook/addon-actions": "^5.3.18",
    "@storybook/addon-links": "^5.3.18",
    "@storybook/addons": "^5.3.18",
    "@storybook/preset-typescript": "^3.0.0",
    "@storybook/react": "^5.3.18",
    "@types/node": "^13.13.2",
    "@types/react": "^16.9.34",
    "babel-loader": "^8.1.0",
    "typescript": "^3.8.3"
  }
}
```
![19](https://user-images.githubusercontent.com/31827190/80140193-34989100-85da-11ea-9f02-9de8ec54e5e2.PNG)
![18](https://user-images.githubusercontent.com/31827190/80140176-306c7380-85da-11ea-9099-c0c7bdc77a07.PNG)

**Comment by styfle** [comment]
It looks like your override for the Build Command doesn't build Next.js.

You could try changing your build command to build both Next.js and Storybook.

Try `yarn build-storybook && yarn build` or `yarn build && yarn build-storybook`.

Related to #3774.

**Reply by jigz** [reply]
I've tried this, I believe it's looking for `routes-manifest.json` inside `storybook-static` output directory rather than next's build directory.

One thing I haven't tried yet, is to just build a next app that will display the storybook build, which sounds counter productive :laughing:

**Reply by styfle** [reply]
Do you want to use Next.js and Storyblock? Or only Storyblock?

If you want both. I think you need to set the Output Directory to `public` and then change `build-storybook` to `build-storybook -o public` so that both Next.js and Storyblock output to the `public` directory. This might have undefined behavior if any of the output files are the same.

**Reply by jigz** [reply]
only Storybook, I think it's risky to merge next and storybook's Output Directory like that

**Reply by styfle** [reply]
It looks like our automatic framework detection is finding the `next` dependency in your `package.json` and assuming you want to build a Next.js website.

May I ask why you have `"next": "9.3.5"` in your dependencies if you don't intend to use the output?

**Reply by jigz** [reply]
I used the next template and added storybook to kickstart this..
though I don't need next at the moment, this component library is meant to be part of a bigger project. I'd prefer not to have a separate repo for storybook and one for the next app.

I tried several preset framework settings, to no avail.

**Reply by styfle** [reply]
Vercel doesn't work well with monorepos yet but we are working on a proper solution.

I think the best case solution for you today is to change the Framework preset to Next.js, then remove the overrides, then update package.json build script to the following:

```
"build": "next build && build-storybook -o public/storybook",
```

**Reply by jigz** [reply]
Special thanks to Samuel Ko @samsisle 

It deployed successfully after updating the build command and adding `now.json` file on root directory. 
`"build": "next build && build-storybook"`
``` 
//now,json
{
  "builds": [
    {
      "src": "package.json",
      "use": "@now/static-build",
      "config": { "distDir": "storybook-static" }
    }
  ]
}
```

**Reply by styfle** [reply]
We fixed this bug so you can remove your `now.json` file and use the Build Command in the dashboard 👍

**Reply by manas-mesh** [reply]
I am still facing this issue with the below project config.

<img width="817" alt="Screenshot 2022-07-07 at 4 22 14 AM" src="https://user-images.githubusercontent.com/105203423/177656377-88ed2244-6914-4284-a6f5-88e3b7031b87.png">
<img width="817" alt="Screenshot 2022-07-07 at 4 28 16 AM" src="https://user-images.githubusercontent.com/105203423/177656921-3a430fb7-671e-430f-a6a7-60d39ed0b71f.png">

**Reply by styfle** [reply]
@manas-mesh You need to change the Framework Preset from "Next.js" to "Other"

**Reply by manas-mesh** [reply]
I've tried it with "Other" as well, getting same error message.

**Reply by manas-mesh** [reply]
@styfle 
<img width="817" alt="Screenshot 2022-07-07 at 5 37 56 AM" src="https://user-images.githubusercontent.com/105203423/177662689-e0b63d20-3d59-4baf-a93f-516d07cc0ac8.png">
<img width="817" alt="Screenshot 2022-07-07 at 5 37 33 AM" src="https://user-images.githubusercontent.com/105203423/177662700-f6b29895-ba22-4600-a7a7-a29b2bc2d9c0.png">

The workaround that I am using is building the storybook locally and pushing the `storybook-static` directory and then in project config setting the root directory as `storybook-static`, but yeah this seems counter productive.

**Reply by styfle** [reply]
@manas-mesh  Do you have `vercel.json` or `now.json` referencing next?

**Reply by manas-mesh** [reply]
@styfle no I don't

**Reply by styfle** [reply]
Can you share the source code or perhaps create a small repository to reproduce the issue?

**Comment by leerob** [comment, marked as answer]
I wrote a guide to make deploying Storybook to Vercel easier. Hopefully this helps, let me know if you have further questions.

https://vercel.com/guides/storybook-with-vercel
